src/core/authorization_request/verification/did.rs

```diff
@@ -6,17 +6,29 @@ use crate::core::{
 use anyhow::{bail, Context, Result};
 use base64::prelude::*;
 use serde_json::{Map, Value as Json};
-use ssi::did_resolve::{resolve_key, DIDResolver};
+use ssi::{
+    dids::{DIDResolver, VerificationMethodDIDResolver},
+    jwk::JWKResolver,
+    verification_methods::{
+        GenericVerificationMethod, InvalidVerificationMethod, MaybeJwkVerificationMethod,
+        VerificationMethodSet,
+    },
+};
 
 /// Default implementation of request validation for `client_id_scheme` `did`.
-pub async fn verify_with_resolver(
+pub async fn verify_with_resolver<M>(
     wallet_metadata: &WalletMetadata,
     request_object: &AuthorizationRequestObject,
     request_jwt: String,
     trusted_dids: Option<&[String]>,
-    resolver: &dyn DIDResolver,
-) -> Result<()> {
-    let (headers_b64, _, _) = ssi::jws::split_jws(&request_jwt)?;
+    resolver: &VerificationMethodDIDResolver<impl DIDResolver, M>,
```

Comment:
Not sure if this is the correct resolver type to use here.

Possibly change to `impl JWKResolver` type?

Reply:
Yep, you can remove the associated type and use:

```rust
    pub async fn new(
        vm: String,
        signer: Arc<dyn RequestSigner + Send + Sync>,
        resolver: &VerificationMethodDIDResolver<
            impl DIDResolver,
            impl MaybeJwkVerificationMethod
                + VerificationMethodSet
                + TryFrom<GenericVerificationMethod, Error = InvalidVerificationMethod>,
        >,
```